app/celery/research_mode_tasks.py

```diff
@@ -22,21 +28,8 @@
 
 
 def send_sms_response(provider, reference, to):
```

Comment:
If you want the callback to find the notification afterwards, the `reference` field must be filled in the database.

See what's done for "normal" SMS
https://github.com/cds-snc/notification-api/blob/ab6e86f452e00b84e28388633ad2da30819cda3a/app/delivery/send_to_providers.py#L88-L90 or for emails in research mode https://github.com/cds-snc/notification-api/blob/ab6e86f452e00b84e28388633ad2da30819cda3a/app/delivery/send_to_providers.py#L179-L182

Can you write a test to make sure that the reference value is filled and is set correctly in the SNS payload passed to Celery?

Reply:
Yes. I will double check as well to see which reference is getting written in the database....

Reply:
Revisiting this comment, there might be some confusion and I just want to clarify. The `reference` field is filled in the database for SMS and when in research mode or with a test API key. The code to make it happen is a bit above the  previous one mentioned:
https://github.com/cds-snc/notification-api/blob/ab6e86f452e00b84e28388633ad2da30819cda3a/app/delivery/send_to_providers.py#L70-L72